src/App.tsx

```diff
@@ -15,48 +16,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface Params {
+  sortField: SortType,
+  reverseField: ReverseType,
+}
+
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  empty = '',
+}
+
+enum ReverseType {
+  reverse = 'reverse',
+  empty = '',
+}
+
+function getPreparedGoods(
+  goods: string[], { sortField, reverseField }: Params,
+): string[] {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+function resetField(field: (param: string) => void) {
+  field('');
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reverseField } as Params,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': reverseField !== ReverseType.reverse,
```

Comment:
Need to write a helper function for this logic `reverseField !== ReverseType.reverse` and use them everywhere
```suggestion
            'is-light': reverseField !== ReverseType.reverse,
```